src/main/java/edu/kis/vh/nursery/FIFORyhmer.java

```diff
@@ -17,14 +19,17 @@ public FIFORyhmer() {
     public int countOut() {
         while (!callCheck())
 
-            temp.countIn(super.countOut());
+            temp.push(super.countOut());
 
-        final int ret = temp.countOut();
+        final int ret = temp.pop();
 
-        while (!temp.callCheck())
+        while (!temp.isEmpty())
 
-            countIn(temp.countOut());
+            countIn(temp.pop());
 
         return ret;
     }
 }
+
+//Best option is IntArrayStack as in temp we only need the latest
+//value so less memory is being occupied
```

Comment:
No, temp pushes and later pops all values. If Ryhmer has more values than temps capacity the operation will fail. Implementation without limitations is needed (IntLinkedlist)